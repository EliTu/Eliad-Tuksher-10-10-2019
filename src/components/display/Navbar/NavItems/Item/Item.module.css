@import '../../../../../variables.css';

.ItemStyles {
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: var(--text-size-medium);
    margin: 0 auto;
    color: var(--color-main-grey-dark);
    border-bottom: 5px solid;
    border-color: rgba(255, 255, 255, 0);
    height: 100%;
    width: 50%;
    transition: all 0.3s ease-in;
}

.ItemStyles a {
    text-decoration: none;
}

.ItemStyles:hover {
    background-color: rgba(82, 159, 223, 0.623);
}

.active {
    border-bottom-color: var(--color-main-black);
}