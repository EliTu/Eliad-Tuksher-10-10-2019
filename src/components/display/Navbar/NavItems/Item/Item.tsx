import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Item.module.css';

interface IProps {
	children: string;
	link: string;
}

const Item: React.FC<IProps> = ({ children, link }) => {
	const { ItemStyles, active } = styles;

	return (
		<NavLink
			className={ItemStyles}
			to={link}
			activeClassName={active}
			exact
		>
			{children}
		</NavLink>
	);
};

export default Item;
