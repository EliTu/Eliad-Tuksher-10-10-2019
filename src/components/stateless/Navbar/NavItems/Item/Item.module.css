@import '../../../../../variables.css';

.ItemStyles {
    font-size: var(--text-size-medium);
    margin: 2rem;
}