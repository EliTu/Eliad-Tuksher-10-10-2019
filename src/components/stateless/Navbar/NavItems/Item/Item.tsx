import React from 'react';
import styles from '.Item.module.css';

const Item: React.FC = () => {
    const { ItemStyles } = styles;
    
	return <div className={ItemStyles}></div>;
};

export default Item;
