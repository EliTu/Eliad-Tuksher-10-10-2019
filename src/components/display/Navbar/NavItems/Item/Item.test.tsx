import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Item from './Item';
import '../../../../../setupTests';
import initialShallowRender from '../../../../../utilities/test-utilities/initialShallowRender';

describe('Item component', () => {
	let component: ShallowWrapper;
	beforeEach(() => {
		component = shallow(<Item link="/s">Test Item</Item>);
	});

	it('should render without errors', () =>
		initialShallowRender(component, '.ItemStyles'));

	it('should render a children prop with a type of string without errors', () => {
		expect(component.children().text()).toBe('Test Item');
	});
});
