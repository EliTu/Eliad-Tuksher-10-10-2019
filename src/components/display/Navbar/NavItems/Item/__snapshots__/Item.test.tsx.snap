// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should render without errors 1`] = `
<NavLink
  activeClassName="active"
  className="ItemStyles"
  exact={true}
  to="/s"
>
  Test Item
</NavLink>
`;
